esp-hal-procmacros/src/interrupt.rs

```diff
@@ -96,6 +96,15 @@ pub fn handler(args: TokenStream, input: TokenStream) -> TokenStream {
     quote::quote_spanned!(original_span =>
         #f
 
+        const _: () = {
+            core::assert!(
+            match #priority {
+                #root::interrupt::Priority::None => false,
+                _ => true,
+            },
+            "Priority::None is not supported");
```

Comment:
This makes a lot of sense and it's very simple, but raises the question why Priority::None is exposed to the user in the first place. We don't really have to resolve that part just yet, so this is good enough to go with.